docs/content/sips/000-sqlite.md

```diff
@@ -0,0 +1,139 @@
+title = "SIP 000 - sqlite"
+template = "main"
+date = "2023-04-17:00:00Z"
+---
+
+Summary: Provide a generic interface for access to a sqlite databases
+
+Owner(s): ryan.levick@fermyon.com
+
+Created: Apr 17, 2023
+
+## Background
+
+Spin currently supports two database types: mysql and [postgres](https://developer.fermyon.com/cloud/data-postgres) which both require the user to provide their own database that is exposed to users through the SDK. This is largely a stopgap until sockets are supported in wasi and there is no longer a need for bespoke clients for these databases (users can bring their favorite client libraries instead).
+
+In contrast to the these other interfaces, the sqlite implementation would easily allow local Spin deployment to use a local sqlite database file, and it provides those hosting Spin deployment envionments (e.g., Fermyon Cloud) to implement lightweight sqlite implementations. In short, a sqlite interface in Spin would allow for a "zero config" experience when users want to work with a SQL database.
+
+### What about `wasi-sql`?
+
+[`wasi-sql`](https://github.com/WebAssembly/wasi-sql) is a work-in-progress spec for a generic SQL interface that aims to support "the features commonly used by 80% of user application". It is likely that when `wasi-sql` is more mature users will be able to successfully use functionality based on the `wasi-sql` interface to interact with a sqlite databases. However, there are still reasons that a dedicated sqlite interface would still be useful:
+
+* For the 20% of use cases where `wasi-sql` is too generic, a dedicated `sqlite` interface can provide that functionality. 
+* The `wasi-sql` spec is under active investigation, and there are large questions about how to best support such a wide breadth of sql flavors. This implementation can help clarify those questions and push upstream work further along.
```

Comment:
For me the question is: do we want to use the opportunity to help _make_ `wasi-sql` ready?

Specifically, I'm thinking it would be nice to follow the precedent we set with key-value such that you can run the same code locally and in Cloud with only a runtime config change.  And since that's kind of the point of `wasi-sql`, we could implement it and making any changes required as necessary.

I'm wary about adding interfaces which can only practically be implemented locally, so I'd either like to see something more like `wasi-sql` or a plan for providing a "cloud-scale" SQLite service.

Reply:
To add on top of what Radu says: `wasi-sql` is a long way from stabilization. We can use this work to help inform the evolution of `wasi-sql`, but there are a lot of open questions about the viability of `wasi-sql` in general since sql flavors differ very substantially. Focusing on one sql flavor, sqlite, will make our lives a lot easier and unblock things much more quickly than trying to conform to a future `wasi-sql` standard.